Heiser analysis figures

nbedanova opened the pull request:
@aarmey Added a few figures to look at deviance transform factorization. When importing the Heiser data using the import_Heiser() function, you can include argument import_Heiser(deviance=True) to use deviance transformation code to prepare the dataset. Default is false and will just do our typical normalization. 

"/home/nicoleb/'C3TAg_50" Is the factorization with our typical normalization with rank 50.

nbedanova commented on pf2rnaseq/imports.py:
```diff
@@ -4,12 +4,55 @@
+    # Remove cells and genes with fewer than 10 reads
+    # X = X[X.X.sum(axis=1) > 10, X.X.sum(axis=0) > 10]
+    readmean, _ = mean_variance_axis(X.X, axis=0)  # type: ignore
+    X = X[:, readmean > geneThreshold]
```
Used same gene threshold cutoff I previously have used to keep dataset being factorized the same.

aarmey commented on pf2rnaseq/figures/figureHeiser1.py:
```diff
@@ -30,16 +30,15 @@ def samples_only(X) -> pd.DataFrame:
-    X = pf2(X, 100)
-
-    X.uns["Pf2_A"] = correct_conditions(X)
+    X=read_h5ad("/home/nicoleb/'C3TAg_50")
```
Are you supposed to have a ' in the filename here?

nbedanova replied:
Yes the filename does have an apostrophe, that's my bad need to rename the saved file.

aarmey commented on pf2rnaseq/figures/figureHeiserGenePac.py:
```diff
@@ -0,0 +1,24 @@
+"Plot gene pacmap"
```
Docstrings use triple quotes """

aarmey commented on pf2rnaseq/figures/figureHeiserFMSDev.py:
```diff
@@ -0,0 +1,186 @@
+    ranksList: list[int],
+    runs: int,
+):
+    # Plots FMS when using different Pf2 components comparing the deviance transformed data vs how we usually normalize
```
Shouldn't this be able to reuse code that calculates the FMS?

nbedanova replied:
I rewrote this as I was thinking to calculate FMS between the factorization results using the deviance transform vs our previous normalization method, not by bootstrapping. In hindsight not sure if this will actually tell us much.

aarmey commented on pf2rnaseq/figures/figureHeiserFMSDev.py:
```diff
@@ -0,0 +1,186 @@
+    # Plots FMS when using different Pf2 components
+    fmsLists = []
+
+    for j in range(0, runs, 1):
```
Seems like this should go in a non-figure file. Also, separate calculation from plotting.

aarmey commented on pf2rnaseq/figures/figureHeiserFMSDev.py:
```diff
@@ -0,0 +1,186 @@
+    dataX = pf2(X, rank, doEmbedding=False)
+
+    fmsLists = []
+
```
Seems like this should go in a non-figure file. Also, separate calculation from plotting.

aarmey commented on pf2rnaseq/imports.py:
```diff
@@ -1,13 +1,55 @@
+
+    non_y_ij = n_i[:, None] - y_ij
+    mu_ij = n_i[:, None] * pi_j[None, :]
+    signs = np.sign(y_ij - mu_ij[None, :])
```
@nbedanova I had an error in my code here. I thought I had passed along the fix, but apparently not. This is now correct, so see if it changes the results.

nbedanova replied:
Ah ok will redo analysis, thanks!

aarmey commented on pf2rnaseq/imports.py:
```diff
@@ -1,13 +1,55 @@
+    X.obs["condition_unique_idxs"] = X.obs["condition_unique_idxs"].astype("category")
+
+    # Pre-calculate gene means
+    X.var["means"] = np.zeros(X.shape[1])
```
There shouldn't be a need to subtract off the mean with this transformation—it is already centered at the MLE.

nbedanova commented:
Fixed suggested changing including moving FMS calculation away out of figure files. Renamed the 50 rank factorization to "/home/nicoleb/C3TAg_50" if needed. Fixed docstring. Deleted some files that are from previous analysis that I have not used (ie. CITEseq files).
